src/util/resize.js

```diff
@@ -0,0 +1,49 @@
+export const DEFAULT_VERTICAL_FLEX = ['1', '1', '1'];
+export const DIVIDER_HEIGHT = 4;
+export const EDITOR_MIN_HEIGHT = 85;
+
+function isSecondDividerPushingUpFirstDivider(height0, height1, deltaY) {
+  return height0 > EDITOR_MIN_HEIGHT &&
+    height1 === EDITOR_MIN_HEIGHT &&
+    deltaY < 0;
+}
+
+function adjustY(height2, lastY, deltaY) {
+  const maxDeltaY = height2 - EDITOR_MIN_HEIGHT;
+  return lastY + Math.min(deltaY, maxDeltaY);
+}
+
+export function updateVerticalFlex({deltaY, heights, index, lastY, y}) {
+  const [height0, height1, height2] = heights;
+  if (index === 0) {
+    return [
+      `0 1 ${y}px`,
+      '1',
+      height2 ? `0 1 ${height2}px` : '1',
+    ];
+  }
+  const offset = DIVIDER_HEIGHT + height0;
+  const offsetY = y - offset;
+  if (
+    isSecondDividerPushingUpFirstDivider(height0, height1, deltaY)
+  ) {
+    return [
+      `0 1 ${height0 + deltaY}px`,
+      `0 1 ${offsetY}px`,
+      '1',
+    ];
+  } else if (height2 > EDITOR_MIN_HEIGHT) {
+    return [
+      `0 1 ${height0}px`,
+      `0 1 ${adjustY(height2, lastY, deltaY) - offset}px`,
+      '1',
+    ];
+  } else if (deltaY < 0 && offsetY <= height1) {
```

Comment:
You're correct about what each condition is testing, but there's an extra bit to consider: what happens when you drag the divider down as far as it will go, and you keep going with the mouse? 

Without some form of correction for this, what happens is the height of the middle container gets inflated, and the top container will suddenly shrink. You can see it for yourself in the original js codepen that you sent me http://codepen.io/chriscoyier/pen/pJpbWK (drag the second divider past the min height of the right section, then drag it back to the middle without letting go).

I can leave comments on these conditions if you want? I know the general project style is to favor code clarity over comments though, which I'm a big fan of.